api/api/views/viewsets/search.py

```diff
@@ -0,0 +1,64 @@
+# Search across all models
+
+from rest_framework import viewsets
+from rest_framework.response import Response
+from rest_framework import status
+from rest_framework.permissions import IsAuthenticated
+from django.db.models import Q
+
+from api.models import (
+    Pentester,
+    Team,
+    Manager,
+    Pentester
+)
+from api.models.mission import Mission
+from api.models.vulns import Vulnerability
+
+
+class SearchView(viewsets.ViewSet):
+    permission_classes = [IsAuthenticated]
+
+    def search(self, request):
+        query = request.query_params.get('q', None)
+        if query is None:
+            return Response(status=status.HTTP_400_BAD_REQUEST)
+        look_for_team = request.query_params.get('look_for_team', False)
+
+        # Search across all models
+        managers = Manager.objects.filter(
+            Q(first_name__icontains=query) |
+            Q(last_name__icontains=query) |
+            Q(email__icontains=query)
+        )
+        pentesters = Pentester.objects.filter(
+            Q(first_name__icontains=query) |
+            Q(last_name__icontains=query) |
+            Q(email__icontains=query)
+        )
+        teams = Team.objects.none()
+        if look_for_team:
+            teams = Team.objects.filter(
+                Q(name__icontains=query) |
+                Q(description__icontains=query)
+            )
+        vulnerabilities = Vulnerability.objects.filter(
+            Q(name__icontains=query) |
+            Q(description__icontains=query) |
+            Q(impact__icontains=query) |
+            Q(recommendation__icontains=query)
+        )
+        missions = Mission.objects.filter(
+            Q(name__icontains=query) |
+            Q(description__icontains=query) |
+            Q(impact__icontains=query) |
+            Q(recommendation__icontains=query)
+        )
+
+        return Response({
+            'managers': managers,
+            'pentesters': pentesters,
```

Comment:
it would be more convenient for the front-end if you could sort items by the closest match. This can be done trivially like this: 

```python
def get_name_value(self, obj) -> str:
        """
            get value used to compute most recurrent letters

            to do that, we get the important metadata that the user _should_ be
            looking for, & then how many letters are most similar. higher score wins.

        """
        if isinstance(obj, Manager):
            return f'{obj.first_name}{obj.last_name}{obj.email}'
        # you get the idea, ...


 def sort_by_closest_match(self, user_input: str, items: List[Any]) -> List[Any]:
        scores: dict[Any, int] = {}

        for item in items:
            score = 0

            # fetching the relevant metadata that the user would be
            # looking for (spaces or whatever dont matter)
            item_str = self.get_name_value(item) # You will have to implement this

            # all characters should be lowercase for a more accurate score
            if len(user_input) <= len(item_str):
                first_string, second_string = user_input.lower(), item_str.lower()
            else:
                first_string, second_string = item_str.lower(), user_input.lower()

            len_first = len(first_string)
            for ctr in range(len_first):
                score += 1 if first_string[ctr] == second_string[ctr] else 0

            scores[item] = score

        # sort scores by dict value (in reverse), then put in in a new dict
        sorted_scores = dict(sorted(scores.items(), key=lambda item: item[1], reverse=True))

        # just make a list out of the keys
        return list(sorted_scores.keys())
```

It can be even faster with numpy, but i don't have time to look into it right now